nixos/modules/services/misc/gotenberg.nix

```diff
@@ -0,0 +1,257 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.gotenberg;
+
+  args =
+    [
+      "--api-port=${toString cfg.port}"
+      "--api-timeout=${cfg.timeout}"
+      "--api-root-path=${cfg.rootPath}"
+      "--log-level=${cfg.logLevel}"
+      "--chromium-max-queue-size=${toString cfg.chromium.maxQueueSize}"
+      "--libreoffice-restart-after=${toString cfg.libreoffice.restartAfter}"
+      "--libreoffice-max-queue-size=${toString cfg.libreoffice.maxQueueSize}"
+      "--pdfengines-engines=${lib.concatStringsSep "," cfg.pdfEngines}"
+    ]
+    ++ optional (cfg.enableBasicAuth) "--api-enable-basic-auth"
+    ++ optional (cfg.chromium.autoStart) "--chromium-auto-start"
+    ++ optional (cfg.chromium.disableJavascript) "--chromium-disable-javascript"
+    ++ optional (cfg.chromium.disableRoutes) "--chromium-disable-routes"
+    ++ optional (cfg.libreoffice.autoStart) "--libreoffice-auto-start"
+    ++ optional (cfg.libreoffice.disableRoutes) "--libreoffice-disable-routes";
+
+  inherit (lib)
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    types
+    mkIf
+    optional
+    optionalAttrs
+    ;
+in
+{
+  options = {
+    services.gotenberg = {
+      enable = mkEnableOption "Gotenberg, a stateless API for PDF files";
+
+      package = mkPackageOption pkgs "gotenberg" { };
+
+      # Users can override these if they need to(ungoogled-chromium, different LO version, etc)
+      # Don't allow setting the qpdf, pdftk, or unoconv paths, as those are very stable
+      # and there's only one version of each.
+      libreofficePackage = mkPackageOption pkgs "libreoffice" { };
+      chromiumPackage = mkPackageOption pkgs "chromium" { };
```

Comment:
I think you forgot to move these inside libreoffice/chromium settings in the last round.

```suggestion
```